### DISCUSSION TITLE
Socket Error on Read Socket

### DISCUSSION THREAD

**Discussion by manuspam** [rhubarbgroup/.github#16] [Support (ROC)] [answered]
I keep getting "Error establishing a Redis connection", specifically "socket error on read socket" errors.
However, Redis Status is Connected, Filesystem is Writeable, and Redis is Reachable in Redis Object Cache WP Plugin.
The connection errors seem to pop up especially with new Woocommerce orders, creating new WP users or existing users logging in.
I can log in as such failed Wordpress user and still see the error.
Flushing cache AFTER such actions seems to clear things out. But this is not an option, as each such error creates a customer complaint after an unsuccessful Woocommerce order.
I could not find any help in the documentation about this specific error.
Any help on this one?

**Comment by tillkruss** [comment, marked as answer]
I'd suggest monitoring Redis Server to make sure you're not running out of connections, that would cause `socket error on read socket`.

Are you using any kind of Redis monitoring tool?

**Reply by manuspam** [reply]
Can you define: "Running out of connections"? Under which circumstances could this happen?
My server is quite loaded with some plugins, hence slow, despite a fast cpu. Could that be an issue?
I just use your own free plugin which offers limited monitoring options. Is there a free monitoring alternative?

It has to do something with user management.
I just got this error again when switching users, with the following URL:
https://www.myserver.com/members/member-name/?user_switched=true
The error will only disappear when erasing browser cache to log out, then erasing redis cache.

**Reply by tillkruss** [reply]
I mean Redis Server running out of connections because too many connections from PHP are open.
